### PR TITLE
The cooler execute

### PR DESCRIPTION
**Pull request by gingershaped**
Add a JS function to pass more detailed settings to the interpreter

### CODE REVIEW

File: js/src/vyxal/JSVyxal.scala [hunk 1]
```diff
@@ -54,6 +54,45 @@ object JSVyxal:
...
+      endPrintMode: EndPrintMode = EndPrintMode.Default,
+      defaultValue: String = "0",
+      rangify: Boolean = false,
+      rangeStart: Int = 1,
+      rangeOffset: Int = 0,
+      literate: Boolean = false,
+  ): Unit =
+    val inputList =
+      inputs.map(MiscHelpers.eval(_)(using Context())).toSeq.reverse
+    val settings = Settings(
```
Comment (AMiller42) [inline js/src/vyxal/JSVyxal.scala:67]: Maybe I'm just not understanding, but what's the benefit of this over parsing a list of flags like in the old execute?

ex. `val settings = Settings(online = true).withFlags(flags.toList)`

Comment (gingershaped) [reply js/src/vyxal/JSVyxal.scala:67]: It maps better to the UI design I have in mind; beyond that, it gives the frontend more power to configure Vyxal as it wishes.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -44,3 +44,11 @@ jobs:
       - uses: coursier/cache-action@v6
       - name: Compile
         run: sbt vyxalJS/compile
+      - name: Link
+        run: sbt vyxalJS/fastLinkJS
```
Comment (lyxal) [inline .github/workflows/test.yaml:48]: Why not `fullLinkJS`?

Comment (gingershaped) [reply .github/workflows/test.yaml:48]: `fastLinkJS` is faster, and these aren't production builds; they're mostly for checking that the code does what you want without compiling it yourself

**Review by lyxal — CHANGES_REQUESTED** [review]
Testing to see if the bot posts this

**Pull request closed without merging**